contracts/YCreditStrat.sol

```diff
@@ -0,0 +1,80 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.7.3;
+
+import "./IStrat.sol";
+import "./IYCredit.sol";
+import "./IVault.sol";
+import "./Timelock.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+contract YCreditStrat is IStrat {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    IVault public vault;
+    IYCredit public yCredit;
+    Timelock public timelock;
+
+    modifier onlyVault {
+        require(msg.sender == address(vault));
+        _;
+    }
+
+    constructor(IVault vault_, Timelock timelock_) {
+        vault = vault_;
+        timelock = timelock_;
+        yCredit = IYCredit(address(vault_.underlying())); // vault underlying must be yCredit
+    }
+
+    function invest() external override onlyVault {
+        reinvest();
+        uint balance = yCredit.balanceOf(address(this));
+        require(balance > 0);
+        yCredit.stake(balance);
+    }
+
+    function divest(uint amount) external override onlyVault {
+        reinvest();
+        yCredit.unstake(amount);
+        yCredit.transfer(address(vault), amount);
```

Comment:
Also here, but more understandable. We can always check the balance after receiving rewards and then withdraw using that balance. But it's more risky, I don't feel strongly.

Reply:
I was thinking we need to determine the amount sent by the vault before it called `invest`. But in fact, it's not needed. Will remove the first `oldBalance` call.